numba/array_analysis.py

```diff
@@ -1422,7 +1432,59 @@ def _analyze_op_call_numpy_reshape(self, scope, equiv_set, args, kws):
             typ = self.typemap[args[1].name]
             if isinstance(typ, types.BaseTuple):
                 return args[1], []
-        return tuple(args[1:]), []
+
+        # Reshape is allowed to take one argument that has the value -1.
+        # This means that the size of that dimension should be inferred from
+        # the size of the array being reshaped and the other dimensions
+        # specified.  Our general approach here is to see if the reshape
+        # has any -1 arguments.  If it has more than one then throw a
+        # ValueError.  If exactly one -1 argument is found, remember its
+        # argument index.
+        stmts = []
+        neg_one_index = -1
+        for arg_index in range(1, len(args)):
+            reshape_arg = args[arg_index]
+            reshape_arg_def = guard(get_definition, self.func_ir, reshape_arg)
+            if isinstance(reshape_arg_def, ir.Const):
+                if reshape_arg_def.value == -1:
+                    if neg_one_index == -1:
+                        neg_one_index = arg_index
+                    else:
+                        msg = ("The reshape API may only include one -1 argument.")
+                        raise errors.ApiError(msg, loc=reshape_arg.loc)
```

Comment:
I think @seibert had a good name for this error class but I don't recall what it was (something user-defined-error related).

Reply:
Okay.  Back to value error with loc info in the message.